open-api/rest-catalog-open-api.yaml

```diff
@@ -3082,6 +3132,47 @@ components:
         default-sort-order-id:
           type: integer
 
+    AssertOverwriteRows:
+      description:
+        This validation helps ensure that overwrite row updates are aligned with the current state of the table.
+        It checks that no conflicting data has been added, deleted, or modified since the operation began. 
+        A conflict detection filter can be applied to identify overlapping data changes, helping to ensure
+        that the update modifies only the intended rows. This validation is optional and may not be required 
+        for certain operations, such as append only table updates
+      required:
+        - type
+      properties:
+        type:
+          type: string
+          enum: [ "assert-overwrite-rows" ]
+        conflict-detection-filter:
```

Comment:
I think having separated requirements like AssertNoConflictWithFilter, AssertNoConflictingDeletes, etc. seems more modular and they could potentially be reused in other update types? Curious what others think.